From bce9dd27f7b080362371e55a87cf299c2e2e225f Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 17 Jul 2012 14:56:34 +0300
Subject: [PATCH 252/311] confc.h: add a note on using confc in FOM state
 functions

c2_confc::cc_group should not point to c2_fom_locality:fl_group.
The documentation of ->cc_group field is extended accordingly.

LogD #466
RB: r/865/diff/1-4/?file=22712#file22712line399
---
 conf/confc.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/conf/confc.h b/conf/confc.h
index 641c1eb..d77c120 100644
--- a/conf/confc.h
+++ b/conf/confc.h
@@ -458,6 +458,13 @@ struct c2_confc {
 	/**
 	 * Serialises configuration retrieval state machines
 	 * (c2_confc_ctx::fc_mach).
+	 *
+	 * Note, that if confc is going to be used in a FOM state
+	 * function, ->cc_group should not point to the request
+	 * handler's c2_fom_locality::fl_group.  Otherwise confc calls
+	 * (e.g., c2_confc_ctx_{init,fini}(), c2_confc_open_sync())
+	 * would deadlock, as the group lock is held when the FOM
+	 * state function is invoked.
 	 */
 	struct c2_sm_group *cc_group;
 	/**
-- 
1.8.3.2

